File: Dsa-day205.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        min_deletions = 0
        b_count = 0
        
        for char in s:
            if char == 'b':
                b_count += 1
            else:  # char == 'a'
                # Either delete this 'a' or delete all previous 'b's
                min_deletions = min(min_deletions + 1, b_count)
        
        return min_deletions

File: Dsa-day205.java
class Solution {
    public int minimumDeletions(String s) {
        int minDeletions = 0;
        int bCount = 0;
        
        for (char c : s.toCharArray()) {
            if (c == 'b') {
                bCount++;
            } else {  // c == 'a'
                // Either delete this 'a' or delete all previous 'b's
                minDeletions = Math.min(minDeletions + 1, bCount);
            }
        }
        
        return minDeletions;
    }
}
